android/java/org/chromium/chrome/browser/crypto_wallet/util/ImageLoader.java

```diff
@@ -5,47 +5,363 @@
+                .listener(new RequestListener<Drawable>() {
+                    @Override
+                    public boolean onLoadFailed(GlideException glideException, Object model,
+                            Target<Drawable> target, boolean isFirstResource) {
+                        return callback != null && callback.onLoadFailed();
+                    }
+
+                    @Override
+                    public boolean onResourceReady(Drawable resource, Object model,
+                            Target<Drawable> target, DataSource dataSource,
+                            boolean isFirstResource) {
+                        return callback != null && callback.onResourceReady(resource, target);
+                    }
+                })
+                .into(imageView);
+    }
+
+    private static BitmapTransformation[] getTransformations(boolean isCircular) {
         if (isCircular) {
-            request = request.circleCrop();
+            return new BitmapTransformation[] {
+                    new FitCenter(), new RoundedCorners(32), new CircleCrop()};
         }
-        return request.priority(Priority.IMMEDIATE).diskCacheStrategy(DiskCacheStrategy.ALL);
+        return new BitmapTransformation[] {new FitCenter(), new RoundedCorners(32)};
     }
 
+    /**
+     * Returns {@code true} if a given URL is supported for download.
+     * @param url Given URL to check
+     * @return {@code true} if a given URL is supported for download, {@code false} otherwise.
+     */
     public static boolean isSupported(String url) {
-        return isValidImgUrl(url) && !isSvg(url);
+        return isValidImgUrl(url);
     }
 
+    /**
+     * Checks if the given URL string represents an SVG image file.
+     * @param url a string representing a URL to an image file.
+     * @return {@code true} if the URL string represents an SVG image file, {@code false} otherwise.
+     */
     public static boolean isSvg(String url) {
+        // Checks if the URL is a valid image URL.
         if (!isValidImgUrl(url)) return false;
+        // Converts the URL to lowercase to make the matching case-insensitive.
+        url = url.toLowerCase(Locale.ENGLISH);
+        // Checks if the URL starts with "data:image/svg" or ends with ".svg".
         return url.startsWith("data:image/svg") || url.endsWith(".svg");
     }
 
+    /**
+     * Checks if the given URL string represents a GIF image file.
+     * @param url a string representing a URL to an image file.
+     * @return {@code true} if the URL string represents an SVG image file, {@code false} otherwise.
+     */
+    public static boolean isGif(String url) {
+        // Checks if the URL is a valid image URL.
+        if (!isValidImgUrl(url)) return false;
+        // Converts the URL to lowercase to make the matching case-insensitive.
+        url = url.toLowerCase(Locale.ENGLISH);
+        // Checks if the URL ends with ".gif".
+        return url.endsWith(".gif");
```

Comment:
Better to avoid straightforward comments to avoid over documentation, no need to document each and every process and step.

Reply:
It's always difficult to find a balance between too much and too little. And also the sweet spot may be subjective.
I'll try to remove obvious comments in favor of a more concise style, but it shouldn't be considered a major issue. In the long run, reading code well documented will make our lives easier.

Reply:
It could take some time to under the difference between well documented and over documented though shouldn't be hard to learn. For e.g. In this case `url.endsWith(".gif");` it is clear that url extension is being checked against `.gif` so no need to explain the same. There are many other method in this PR that doesn't follow this practice for many reasons. Avoiding over documentation would save time of everyone.